hubspot-style/src/main/java/com/hubspot/immutables/utils/WireSafeEnum.java

```diff
@@ -113,6 +116,21 @@ public String asString() {
     return enumValue;
   }
 
+  @Nonnull
+  public T asEnumOrThrow() {
+    return asEnum()
+        .orElseThrow(this::getInvalidValueException);
+  }
+
+  private IllegalStateException getInvalidValueException() {
+    ensureValidMembersCacheInitialized(enumType);
+
+    String message = String.format("Value '%s' is not valid for enum of type '%s'. Valid values are: %s",
+        jsonValue, enumType.getSimpleName(), VALID_MEMBERS_CACHE.get(enumType));
```

Comment:
Can we just compute the valid members each time? Hopefully it's not a hot code path, and it should be dwarfed by the cost of the exception creation